### PR TITLE
[Kafka] New custom input package

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

This adds custom Kafka input package, allowing users to ingest data from kafka using our input, with their own pipelines.

This has been built in the same format and principle as our other input packages.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by P1llus** [comment]
Should work just fine, but I want to see if we can add some kafka system tests, looking for some proper docker images to do this first.

File: packages/kafka_log/data_stream/generic/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,252 @@
+title: Custom Kafka Logs
+type: logs
+streams:
+  - input: kafka
+    description: Collect data from kafka topic with Elastic Agent.
```
Comment (efd6) [inline packages/kafka_log/data_stream/generic/manifest.yml:5]: Upper case "Kafka"?

File: packages/kafka_log/data_stream/generic/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+      - name: password
+        type: password
+        title: Password
+        description: Password used for SASL authentication.
+        required: false
+        show_user: true
+      - name: kerberos_enabled
+        type: bool
+        title: Kerberos Enabled
+        description: The enabled setting can be used to enable the kerberos configuration by setting it to true. The default value is false.
```
Comment (efd6) [inline packages/kafka_log/data_stream/generic/manifest.yml:94]: Upper case "Kerberos"?

File: packages/kafka_log/data_stream/generic/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+      - name: kerberos_keytab
+        type: text
+        title: Kerberos Keytab
+        description: If you configured keytab for Auth Type, you have to provide the path to the keytab of the selected principal.
+        required: false
+        show_user: false
+      - name: kerberos_service_name
+        type: text
+        title: Kerberos Service Name
+        description: This option can only be configured for Kafka. It is the name of the Kafka service, usually kafka.
```
Comment (efd6) [inline packages/kafka_log/data_stream/generic/manifest.yml:132]: Quote final "kafka"?

**Review by kaiyan-sheng — COMMENTED** [review]
Overall it looks good to me. I think we still need system tests here in order to release it as GA right? 
About the owner, not sure if it's appropriate to change right now but I think eventually `elastic/obs-service-integrations` should own this.

**Comment by jsoriano** [comment]
> About the owner, not sure if it's appropriate to change right now but I think eventually `elastic/obs-service-integrations` should own this.

+1 to this, I think we shouldn't have packages owned by integrations as this is becoming confusing, see https://github.com/elastic/integrations/pull/3202

**Comment by lalit-satapathy** [comment]
> 

Just trying to clarify. This new package is developed by security team but not owned by them?

**Comment by lalit-satapathy** [comment]
@P1llus the changes looks good to me. The name "custom kafka logs" confused me, where I thought as the logs of kafka broker. Where as this is actually  just referring to all the messages in "kafka topic".

**Review by lalit-satapathy — APPROVED** [review]

**Pull request merged**